### PR TITLE
Added support for Ubiquiti UFiber OLT

### PR DESCRIPTION
**Pull request by Martin22**
Please give a short description what your pull request is for

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [x] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 14256`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by Martin22** [comment]
@Jellyfrog 
Why was this PR not included in the new version? I don't understand that some tests don't pass, but it's a bit strange that others where there is only a different version of php pass, I see it more of a bug elsewhere than in the PR in question.

File: includes/discovery/ports/edgeosolt.inc.php [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+unset($port_stats);
+
+$sfp_stats = snmpwalk_group($device, 'ubntSfps', 'UBNT-UFIBER-MIB', 1, $sfp_stats);
+
+$index = 0;
+for ($index = 0; $index < count($sfp_stats); $index++) {
```
Comment (murrant) [inline includes/discovery/ports/edgeosolt.inc.php:8]: Can you change this to use foreach instead?

```php
foreach ($sfp_stats as $index => $stats) {
    $port_stats[$index + 1]['ifDescr'] = $stats['ubntSfpName'];
...
```

**Review by murrant — COMMENTED** [review]
Looks much better, nice work.

Can you please use ubntSfpTable instead of ubntSfpEntry (for all instances of Entry)

Entry is structural, not intended for direct use.

File: includes/discovery/ports/edgeosolt.inc.php [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+unset($port_stats);
+
+$sfp_stats = snmpwalk_group($device, 'ubntSfps', 'UBNT-UFIBER-MIB', 1, $sfp_stats);
```
Comment (murrant) [inline includes/discovery/ports/edgeosolt.inc.php:5]:
```suggestion
$sfp_stats = snmpwalk_group($device, 'ubntSfps', 'UBNT-UFIBER-MIB', 1);
```

File: includes/polling/ports/os/edgeosolt.inc.php [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+unset($port_stats);
+
+$sfp_stats = snmpwalk_group($device, 'ubntSfps', 'UBNT-UFIBER-MIB', 1, $sfp_stats);
```
Comment (murrant) [inline includes/polling/ports/os/edgeosolt.inc.php:5]:
```suggestion
$sfp_stats = snmpwalk_group($device, 'ubntSfps', 'UBNT-UFIBER-MIB');
```

Comment (Martin22) [reply includes/polling/ports/os/edgeosolt.inc.php]: The detection didn't work before, but I found out that I need to use a different oid.  I reverted foreach and added an offset, without which even if I set unset to port_stats, some ports got the wrong id.

File: includes/discovery/ports/edgeosolt.inc.php (hunk 1 above)
Comment (Martin22) [reply includes/discovery/ports/edgeosolt.inc.php]: > 

When using table instead of entry, nothing is detected.

**Comment by Martin22** [comment]
> Looks much better, nice work.
> 
> Can you please use ubntSfpTable instead of ubntSfpEntry (for all instances of Entry)
> 
> Entry is structural, not intended for direct use.

When using table instead of entry, nothing is detected.

**Comment by Martin22** [comment]
> Looks much better, nice work.
> 
> Can you please use ubntSfpTable instead of ubntSfpEntry (for all instances of Entry)
> 
> Entry is structural, not intended for direct use.

Changed. Thank you for your advice.

**Review by Jellyfrog — APPROVED** [review]

**Comment by warren-lcf** [comment]
@Martin22 great job! is it possible to get the ONU's out as their own devices with associated RX/TX/Connection States? Being listed as sensors makes it hard to see RX/TX together for the same ONU

**Pull request merged**